### PR TITLE
Qmaps 497 default map center and restore last location

### PR DESCRIPTION
**Pull request by patfrat**
Change default map center
Store the last location
Restore the last location if present

### CODE REVIEW

File: src/adapters/scene.js [hunk 1]
```diff
@@ -23,6 +25,12 @@ function Scene() {
   } else {
     this.center = [map.center.lng, map.center.lat]
     this.zoom = map.zoom
+    // Fly to last location if exists
+    store.getLastLocation().then((loc) => {
+      if (loc) {
+        this.mb.flyTo({center : [loc.lng, loc.lat], zoom : loc.zoom})
```
Comment (Julien-laville) [inline src/adapters/scene.js]: I prefer set the correct center on map load instead of move to

File: src/adapters/scene.js [hunk 2]
```diff
@@ -87,6 +95,14 @@ Scene.prototype.initMapBox = function () {
 
     this.mb.on('moveend', () => {
       UrlState.replaceUrl()
+      fire(
+        'store_center',
```
Comment (Julien-laville) [inline src/adapters/scene.js]:
```suggestion
       fire('store_center', {
          lng: this.mb.getCenter().lng,
          lat: this.mb.getCenter().lat,
          zoom: this.mb.getZoom()
        })
```

I prefer compact notation

File: src/adapters/store.js [hunk 3]
```diff
@@ -27,6 +31,24 @@ Store.prototype.getAllPois = async function() {
   }
 }
 
+Store.prototype.getLastLocation = async function() {
+  try {
+    return await abstractStore.get(`qmaps_v${version}_last_location`)
+  } catch (error) {
+    fire('error_h' , 'store ' + error)
```
Comment (Julien-laville) [inline src/adapters/store.js]: error is not yet supported, you can use console.error for now

File: src/adapters/store.js [hunk 4]
```diff
@@ -27,6 +31,24 @@ Store.prototype.getAllPois = async function() {
...
+    throw error
+  }
+}
+
+Store.prototype.setLastLocation = async function(loc) {
+  try {
+    return await abstractStore.set(`qmaps_v${version}_last_location`, loc)
+  } catch (error) {
+    fire('error_h' , 'store ' + error)
```
Comment (Julien-laville) [inline src/adapters/store.js]: same here, use console.error

File: src/adapters/scene.js [hunk 5]
```diff
@@ -23,6 +25,12 @@ function Scene() {
   } else {
     this.center = [map.center.lng, map.center.lat]
```
Comment (Julien-laville) [inline src/adapters/scene.js]: you should put stored center here

File: src/adapters/scene.js [hunk 6]
```diff
@@ -11,21 +11,33 @@ import loadImage from '../libs/image_loader'
...
 
 function Scene() {
+  this.currentMarker = null
+  this.popup = new PoiPopup()
+}
+
+Scene.prototype.initScene = async function () { 
   UrlState.registerHash(this, 'map')
-  if(window.hotLoadPoi) {
+  if (window.hotLoadPoi) {
```
Comment (Julien-laville) [inline src/adapters/scene.js]: you should regroup all init position logic and put it inside a method called setupMapPosition or something equivalent.

File: src/adapters/scene.js [hunk 7]
```diff
@@ -87,6 +99,10 @@ Scene.prototype.initMapBox = function () {
 
     this.mb.on('moveend', () => {
       UrlState.replaceUrl()
+      let lng = this.mb.getCenter().lng
+      let lat = this.mb.getCenter().lat
+      let zoom = this.mb.getZoom()
+      fire('store_center', { lng: lng, lat: lat, zoom: zoom })
```
Comment (Julien-laville) [inline src/adapters/scene.js]: you can write it with es6 literal notation {lng, lat, zoom}

File: src/adapters/scene.js [hunk 8]
```diff
@@ -120,7 +136,7 @@ Scene.prototype.fitMap = function(poi, options = {}) {
       this.mb.fitBounds(poi.bbox, {padding : padding, animate : false})
     }
   } else {
-    let flyOptions = {center : poi.getLngLat(), screenSpeed: 1.5, animate:false}
+    let flyOptions = {center : poi.getLngLat(), screenSpeed: 1.5, anigithubmate:false}
```
Comment (Julien-laville) [inline src/adapters/scene.js]: you have misspelled "i write on the bad window" :)

Comment (patfrat) [reply src/adapters/scene.js]: oops

**Pull request merged**